lib/health.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const request = require('request-promise-native');
+
+const hasMatchingMethod = (option, reqMethod) => {
+  return option.methods.some(method => {
+    return method.toLowerCase() === reqMethod.toLowerCase();
+  });
+};
+
+module.exports = endpoints => {
+
+  return (req, res, next) => {
+
+    if (endpoints && endpoints.length) {
+
+      let requests = endpoints.reduce((memo, option) => {
+        // Test the method property against the current req method
+        if (option.methods && (hasMatchingMethod(option, req.method) !== true)) {
+          return memo;
+        }
+        // memoize the endpoints
+        memo.push(request(option.url).promise());
+        return memo;
+      }, []);
```

Comment:
I think this should only happen on GET requests. I would opt to mount the middleware on GET requests, rather than filter the requests in this module. What do you think?